keras/backend/theano_backend.py

```diff
@@ -111,6 +111,13 @@ def dot(x, y):
     return T.dot(x, y)
 
 
+def batch_dot(x, y, axes=None):
+    if axes is None:
+        # behaves like tf.batch_matmul as default
```

Comment:
One thing that we could do to avoid this behavior would be to always force the user to pass `axes` as input. Think about it, without `axes` and different default behaviors, changing backends could crash the code. What do you think?